File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>
#include <math.h>

using namespace std;

int solution(int number, int limit, int power) {
    
    int answer = 0;
    
    vector<int> d_num(number + 1, 1);
    
    for(int i = 2 ; i < d_num.size(); i++){
        for(int j = i; j <= d_num.size() ; j += i)
            d_num[j]++;
    }
    
    for(int i = 1 ; i <= number ; i++){
        if(d_num[i] > limit)
            answer += power;
        else
            answer += d_num[i];
    }
    
    return answer;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
import java.util.*;
import java.util.stream.*;

class Solution {
    
    
    public int getFactors(int n){
        int cnt = 0;
        for(int i = 1 ; i <= (int)Math.sqrt(n) ; i++)
            if(n % i == 0){
                cnt++;
                if(n / i != i)
                    cnt++;
            }
        
        return cnt;
    }
    
    
    public int solution(int number, int limit, int power) {
        return (int)IntStream.range(1, number + 1)
            .map(i -> (getFactors(i) <= limit) ? getFactors(i) : power)
            .sum();
    }
}